Orchestration/OrchestrationService/ModuleConfigurationDeployment.ps1

```diff
@@ -108,322 +108,350 @@ Function New-Deployment {
                 -ArchetypeInstance $archetypeInstanceJson `
                 -ArchetypeInstanceName $ArchetypeInstanceName;
 
-        $moduleConfiguration = `
-            Get-ModuleConfiguration `
-                -ArchetypeInstanceJson $archetypeInstanceJson `
-                -ModuleConfigurationName $moduleConfigurationName `
-                -ArchetypeInstanceName $ArchetypeInstanceName `
-                -Operation @{ "False" = "deploy"; "True" = "validate"; }[$Validate.ToString()];;
+        $allModules = @()
 
-        if ($null -eq $moduleConfiguration) {
-            throw "Module configuration not found for module name: $moduleConfigurationName";
-        }
+        if ([string]::IsNullOrEmpty($ModuleConfigurationName)) {
+            
+            $topologicalSortRootPath = `
+                Join-Path $rootPath -ChildPath 'TopologicalSort';
+            
+            Invoke-Command -ScriptBlock { dotnet build $topologicalSortRootPath --configuration Release --output ./ }
```

Comment:
Should this be part of the bootstrap process? Also, Should we first check for the assembly in the path to see if it exists before rebuilding the assembly again? What should be our logic to refresh an assembly when a code change is made? Should we consider versioning the DLL and checking for the specific version before rebuidling again?